src/gl/metal/Shaders.metal

```diff
@@ -825,3 +825,580 @@ highlightFragmentShader(HFSInput in [[stage_in]], constant HVSUniforms& uniforms
     return float4(uniforms.u_color.xyz, result);
 }
+
+// g_postEffectsVertexShader
+
+struct PEVParams
+{
+  float4 u_screenParams;
+  float4 u_saturation;
+};
+
+struct PEVOutput
+{
+  float2 out_var_TEXCOORD0 [[user(locn0)]];
+  float2 out_var_TEXCOORD1 [[user(locn1)]];
+  float2 out_var_TEXCOORD2 [[user(locn2)]];
+  float2 out_var_TEXCOORD3 [[user(locn3)]];
+  float2 out_var_TEXCOORD4 [[user(locn4)]];
+  float2 out_var_TEXCOORD5 [[user(locn5)]];
+  float out_var_TEXCOORD6 [[user(locn6)]];
+  float4 gl_Position [[position]];
+};
+
+struct PEVInput
+{
+  float3 in_var_POSITION [[attribute(0)]];
+  float2 in_var_TEXCOORD0 [[attribute(1)]];
+};
+
+vertex PEVOutput postEffectsVertexShader(PEVInput in [[stage_in]], constant PEVParams& u_params [[buffer(0)]])
+{
+  PEVOutput out = {};
+  out.gl_Position = float4(in.in_var_POSITION.xy, 0.0, 1.0);
+  out.out_var_TEXCOORD0 = in.in_var_TEXCOORD0;
+  out.out_var_TEXCOORD1 = in.in_var_TEXCOORD0 + u_params.u_screenParams.xy;
+  out.out_var_TEXCOORD2 = in.in_var_TEXCOORD0 - u_params.u_screenParams.xy;
+  out.out_var_TEXCOORD3 = in.in_var_TEXCOORD0 + float2(u_params.u_screenParams.x, -u_params.u_screenParams.y);
+  out.out_var_TEXCOORD4 = in.in_var_TEXCOORD0 + float2(-u_params.u_screenParams.x, u_params.u_screenParams.y);
+  out.out_var_TEXCOORD5 = u_params.u_screenParams.xy;
+  out.out_var_TEXCOORD6 = u_params.u_saturation.x;
+  return out;
+}
+
+// g_postEffectsFragmentShader
+
+constant float _66 = {};
+constant float2 _67 = {};
+constant float2 _68 = {};
+
+struct PEFOutput
+{
+  float4 out_var_SV_Target [[color(0)]];
+};
+
+struct PEFInput
+{
+  float2 in_var_TEXCOORD0 [[user(locn0)]];
+  float2 in_var_TEXCOORD1 [[user(locn1)]];
+  float2 in_var_TEXCOORD2 [[user(locn2)]];
+  float2 in_var_TEXCOORD3 [[user(locn3)]];
+  float2 in_var_TEXCOORD4 [[user(locn4)]];
+  float2 in_var_TEXCOORD5 [[user(locn5)]];
+  float in_var_TEXCOORD6 [[user(locn6)]];
+};
+
+fragment PEFOutput main0(PEFInput in [[stage_in]], texture2d<float> texture0 [[texture(0)]], texture2d<float> texture1 [[texture(1)]], sampler sampler0 [[sampler(0)]], sampler sampler1 [[sampler(1)]])
+{
+  PEFOutput out = {};
+  float4 _80 = texture1.sample(sampler1, in.in_var_TEXCOORD0);
+  float _81 = _80.x;
+  float4 _535;
+  if ((1.0 - (((step(abs(texture1.sample(sampler1, in.in_var_TEXCOORD1).x - _81), 0.0030000000260770320892333984375) * step(abs(texture1.sample(sampler1, in.in_var_TEXCOORD2).x - _81), 0.0030000000260770320892333984375)) * step(abs(texture1.sample(sampler1, in.in_var_TEXCOORD3).x - _81), 0.0030000000260770320892333984375)) * step(abs(texture1.sample(sampler1, in.in_var_TEXCOORD4).x - _81), 0.0030000000260770320892333984375))) == 0.0)
+  {
+    _535 = texture0.sample(sampler0, in.in_var_TEXCOORD0);
+  }
+  else
+  {
+    float4 _534;
+    switch (0u)
+    {
+      default:
+      {
+        float2 _123 = _67;
+        _123.x = in.in_var_TEXCOORD0.x;
+        float2 _125 = _123;
```

Comment:
You didn't need this variable, `_123` isn't used other than this

Reply:
This will have something to do with the way the HLSL shader is written. Lots of inline maths gets expanded into locals like this that do a whole bunch of nothing.